torch/csrc/Module.cpp

```diff
@@ -144,6 +144,13 @@ static PyObject* module;
 
 static THPGenerator* THPDefaultCPUGenerator = nullptr;
 
+bool registerPythonPrivateUse1Hook(py::object hook);
+bool registerPythonPrivateUse1DeviceGuard(py::object guard);
+at::Tensor createEmptyTensor(
+  const std::vector<int64_t>& shape,
+  c10::ScalarType dtype
+);
+
```

Comment:
In my opinion, "torch/csrc/Module.cpp" is a thin glue layer connecting Python and C++, so it is better to move the implementation somewhere else.

@albanD, can we create a separate dic in c10/ to store PrivateUse1 files, just like `c10/cuda` or `c10/xpu`? Then we can move `PythonPrivateUse1HooksInterface` and `PythonPrivateUse1DeviceGuard` there.